File: app-miyou/src/main/java/com/miyou/app/application/dialogue/policy/PromptTemplatePolicy.java
package com.miyou.app.application.dialogue.policy;

public record PromptTemplatePolicy(
	String baseTemplate,
	String defaultPersonaTemplate,
	String commonTemplate,
	String configuredSystemPrompt
) {
	public PromptTemplatePolicy {
		baseTemplate = normalize(baseTemplate);
		defaultPersonaTemplate = normalize(defaultPersonaTemplate);
		commonTemplate = normalize(commonTemplate);
		configuredSystemPrompt = normalize(configuredSystemPrompt);
	}

	private static String normalize(String template) {
		return template == null ? "" : template;
	}
}

File: backend/application/src/main/kotlin/com/miyou/app/application/dialogue/policy/PromptTemplatePolicy.kt
package com.miyou.app.application.dialogue.policy

class PromptTemplatePolicy(
    baseTemplate: String,
    defaultPersonaTemplate: String,
    commonTemplate: String,
    configuredSystemPrompt: String,
) {
    val baseTemplate: String = baseTemplate.normalize()
    val defaultPersonaTemplate: String = defaultPersonaTemplate.normalize()
    val commonTemplate: String = commonTemplate.normalize()
    val configuredSystemPrompt: String = configuredSystemPrompt.normalize()

    private fun String?.normalize(): String = this ?: ""
}
